crates/typst/src/foundations/cast.rs

```diff
@@ -328,13 +330,24 @@ impl CastInfo {
             write!(msg, "{}", found.ty()).unwrap();
         }
 
+        let mut msg = HintedString::new(msg.into());
+
         if let Value::Int(i) = found {
-            if parts.iter().any(|p| p == "length") && !matching_type {
-                write!(msg, ": a length needs a unit - did you mean {i}pt?").unwrap();
+            if !matching_type && parts.iter().any(|p| p == "length") {
+                msg.hint(eco_format!("a length needs a unit - did you mean {i}pt?"));
+            }
+        }
+        if let Value::Str(s) = found {
+            if !matching_type && parts.iter().any(|p| p == "label") {
+                if is_valid_in_label_literal(s) {
+                    msg.hint(eco_format!("use `<{s}>` to create a label or the label function to convert a string to a label"));
```

Comment:
Maybe we can improve this message a bit? I would suggest "use `<{s}>` or `#label("{s}")` to create a label".

(I'm open to further suggestions)

Reply:
I like that! But we should probably drop the hash since this error will appear in code mode.

Reply:
What if the offending expression is not a literal?

Reply:
Fair point. But in any case the current wording imo contains the word "label" too many times.

Reply:
Can we first check if it is string literal? And if so, we can have `label("{s}")` as the hint. Otherwise, we fallback to a general hint.

Reply:
Right now, not all values keep span info. The other solution is to make `FromValue::from_value` take span info, which seems icky.

Reply:
How about ``use the label syntax `<{s}>` or create a label from a string with the `label` function``?

Reply:
I didn't see that the code was already changed. I'm also okay with leaving it as-is, with the `"label({})", s.repr()`

Reply:
Anyway, we could change it to show a different message for non-literals in a future PR, perhaps once Typst moves to a bytecode interpreter.